airbyte-integrations/connectors/source-monday/source_monday/graphql_request_options_provider.py

```diff
@@ -0,0 +1,48 @@
+import json
+import os
+from dataclasses import dataclass
+from typing import Optional, Mapping, Any, Union, MutableMapping
+
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.request_options import InterpolatedRequestOptionsProvider
+from airbyte_cdk.sources.declarative.types import StreamState, StreamSlice
+
+
+@dataclass
+class CustomOptionsProviderMixin:
+    limit: Union[InterpolatedString, str] = None
+
+
+@dataclass
+class GraphQLRequestOptionsProvider(CustomOptionsProviderMixin, InterpolatedRequestOptionsProvider):
+    NEXT_PAGE_TOKEN_FIELD_NAME = "next_page_token"
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        super(GraphQLRequestOptionsProvider, self).__post_init__(options)
+
+        self.limit = InterpolatedString.create(self.limit, options=options)
+        self.name = options.get('name', '').lower()
+
+    def get_schema_root_properties(self):
+        schema_path = os.path.join(os.path.abspath(os.curdir), "source_monday", f"schemas/{self.name}.json")
```

Comment:
we do hardcode the path but do not hardcode the name, that's very weird

Reply:
I have probably took the name of the stream for the name of the source, nevermind